src/reducers/tasksReducer.js

```diff
@@ -32,6 +32,24 @@ export default function (state = initialState, action) {
     case 'POST_TASK_REJECTED':
       return state;
 
+    case 'DELETE_TASK_PENDING':
+      return state;
+
+    case 'DELETE_TASK_REJECTED':
+      return state;
+
+    case 'DELETE_TASK_FULFILLED':
+      const newState = JSON.parse(JSON.stringify(state));
+
+      delete newState.taskMap[action.meta.gid];
+      var ndx = newState.rootTaskGids.indexOf(action.meta.gid);
+      if (ndx > -1) {
+        newState.rootTaskGids.splice(ndx, 1);
+      }
```

Comment:
@ajvb In addition to deleting a task from `taskMap` and attempting to delete its gid from `rootTaskGids`, if it has a non-null `parent_task_gid` then we need to remove its gid from it's parent's `subtask_gids` array field.